modules/dnn/src/legacy_backend.hpp

```diff
@@ -201,7 +201,9 @@ struct BlobManager
                 if (refIt != refCounter.end() && refIt->second == 0)
                 {
                     const Mat& unusedBlob = hostIt->second;
-                    if (unusedBlob.total() >= targetTotal && unusedBlob.total() < bestBlobTotal && unusedBlob.type() == dtype)
+                    // Only reuse blobs of exactly the same size to avoid stateless issues
+                    // This prevents memory from previous computations from affecting current results
+                    if (unusedBlob.total() == targetTotal && unusedBlob.type() == dtype)
```

Comment:
@dkurt Could you check that the new condition is correct and the patch does not hide some out-of-buffer access or initialization issue.

Reply:
> When processing images of different sizes, the DNN backend reused memory blobs that were larger than needed, causing residual data from previous computations to affect the current results. By requiring exact size matching, we ensure that only clean, appropriately-sized memory is reused.

@aditya270520, please make a test that reporoduces this behavior. At least two inference runs expected, not just `blobFromImage`